stdlib/observability/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java

```diff
@@ -139,16 +140,25 @@ public boolean finishSpan(int spanId) {
      * @param spanId   id of the Span
      * @return boolean to indicate if tag was added to the span
      */
-    public boolean addTag(String tagKey, String tagValue, int spanId) {
+    public boolean addTag(String tagKey, String tagValue, int spanId, Context context) {
         if (!enabled) {
             return false;
         }
         ObserverContext observerContext = observerContextList.get(spanId);
-        if (observerContext != null) {
+        if (spanId == -1) {
+            Optional<ObserverContext> obs =   ObservabilityUtils.getParentContext(context);
+            if (obs.isPresent()){
```

Comment:
There are few checkstyle issues. Can you fix them, run a local build and update the PR?

Reply:
okay ayiaa i will see to that..

Reply:
Sure I will check on that